module/proto/gravity/v1/msgs.proto

```diff
@@ -1,237 +1,189 @@
-// (TODO determine this without nondeterminism) This message includes the batch
-// as well as an Ethereum signature over this batch by the validator
-// -------------
-message MsgConfirmBatch {
-  uint64 nonce          = 1;
-  string token_contract = 2;
-  string eth_signer     = 3;
-  string orchestrator   = 4;
-  string signature      = 5;
+message MsgRequestBatchTxResponse {}
+
+// MsgSubmitEthereumSignature submits an ethereum signature for a given
+// validator
+message MsgSubmitEthereumSignature {
+  option (gogoproto.goproto_getters) = false;
+
+  // TODO: can we make this take an array?
+  google.protobuf.Any signature = 1
+      [ (cosmos_proto.accepts_interface) = "EthereumSignature" ];
+  string signer = 2;
 }
 
-message MsgConfirmBatchResponse {}
-
-// MsgConfirmLogicCall
-// When validators observe a MsgRequestBatch they form a batch by ordering
-// transactions currently in the txqueue in order of highest to lowest fee,
-// cutting off when the batch either reaches a hardcoded maximum size (to be
-// decided, probably around 100) or when transactions stop being profitable
-// (TODO determine this without nondeterminism) This message includes the batch
-// as well as an Ethereum signature over this batch by the validator
-// -------------
-message MsgConfirmLogicCall {
-  string invalidation_id    = 1;
+// ContractCallTxSignature is a signature on behalf of a validator for a
+// ContractCallTx.
+message ContractCallTxSignature {
+  bytes invalidation_id = 1
+      [ (gogoproto.casttype) =
+            "github.com/tendermint/tendermint/libs/bytes.HexBytes" ];
   uint64 invalidation_nonce = 2;
-  string eth_signer         = 3;
-  string orchestrator       = 4;
-  string signature          = 5;
+  string ethereum_signer = 3;
+  bytes signature = 4
+      [ (gogoproto.casttype) =
+            "github.com/ethereum/go-ethereum/common/hexutil.Bytes" ];
 }
 
-message MsgConfirmLogicCallResponse {}
+// BatchTxSignature is a signature on behalf of a validator for a BatchTx.
+message BatchTxSignature {
+  string token_contract = 1;
+  uint64 nonce = 2;
+  string ethereum_signer = 3;
+  bytes signature = 4
+      [ (gogoproto.casttype) =
+            "github.com/ethereum/go-ethereum/common/hexutil.Bytes" ];
+}
 
-// EthereumBridgeDepositClaim
-// When more than 66% of the active validator set has
-// claimed to have seen the deposit enter the ethereum blockchain coins are
-// issued to the Cosmos address in question
-// -------------
-message MsgDepositClaim {
-  uint64 event_nonce    = 1;
-  uint64 block_height   = 2;
-  string token_contract = 3;
-  string amount         = 4 [
-    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
-    (gogoproto.nullable)   = false
-  ];
-  string ethereum_sender = 5;
-  string cosmos_receiver = 6;
-  string orchestrator    = 7;
+// UpdateSignerSetTxSignature is a signature on behalf of a validator for a
+// UpdateSignerSetTx
+message UpdateSignerSetTxSignature {
```

Comment:
Can we change `UpdateSignerSetTx` to `SignerSetTx`? The verb on the front is confusing IMO